devdata/create_devdata.py

```diff
@@ -0,0 +1,163 @@
+        g.add(
+            (
+                named_graph_uri,
+                FOAF.primaryTopic,
+                URIRef(f"https://data.ons.gov.uk/datasets/{dataset_id}"),
+            )
+        )
+        g.add(
+            (
+                named_graph_uri,
+                DCTERMS.created,
+                Literal(current_datetime, datatype=XSD.dateTime),
+            )
+        )
+
+        # Add an appropriate context to our jsonld to allow converstion to RDF
+        dataset_dict = set_context(dataset_dict)
+
+        # Add the graph of jsonld sourced triples to the named graph
+        # alongside our structural triples
+        jsonld_as_graph = Graph()
+        jsonld_as_graph.parse(data=json.dumps(dataset_dict), format="json-ld")
+        g += jsonld_as_graph
+
+        # Write out our graph
+        out_path = Path(repo_root / f"devdata/out/dataset{i}.ttl")
+        g.serialize(out_path, format="ttl")
+
+        # Now do some (uber primitive) wrangling to get the ttl
+        # into the trig format with the individual named graphs
+        # explicitly shown.
+        with open(out_path) as f:
+            started_graph = False
+            for i, line in enumerate(f.readlines()):
+                if line.startswith("@prefix"):
+                    if line not in prefixes:
+                        prefixes.append(line)
+                else:
+                    if not started_graph:
+                        wrangled_lines.append("")
+                        wrangled_lines.append(f"<{named_graph_uri}>" + " {" + linesep)
+                        started_graph = True
+                    if line.strip() == "":
+                        wrangled_lines.append(line)
+                    else:
+                        wrangled_lines.append("\t" + line)
+            wrangled_lines.append("}" + linesep + linesep)
+
+    # ------------------
+    # Editions resources
+    # ------------------
+    # TODO
+
+    # ------------------
+    # Versions resources
+    # ------------------
+    # TODO
+
+    # etc etc...
+
+    seed = Path(repo_root / "devdata/out/seed.trig")
+    with open(seed, "w") as f:
+        for prefix in prefixes:
+            f.write(prefix)
+        f.write(linesep)
+
+        for line in wrangled_lines:
+            f.write(line)
+
+    # Now get this seed data into the database
+    # adapted from:
+    # - https://github.com/rossbowen/poc-api/blob/main/api/src/poc_api/database.py
+    # - https://github.com/rossbowen/poc-api/blob/main/api/src/poc_api/seed/seed.py
+
+    def skolemise(node):
+        if isinstance(node, BNode):
+            return "<bnode:b%s>" % node
+        return _node_to_sparql(node)
+
+    configuration = (f"{oxigraph_url}/query", f"{oxigraph_url}/update")
+    db = Dataset(store=SPARQLUpdateStore(*configuration, node_to_sparql=skolemise))
+    db.parse(seed)
+
+
+if __name__ == "__main__":
+    oxigraph_url = os.getenv("GRAPH_DB_URL", None)
```

Comment:
When running `pipenv run python3 ./devdata/create_devdata.py` I was getting an issue around getting the "GRAPH_DB_URL" var, unless I missed a setup step.
`AssertionError: You need to specfiy the oxigraph url via the OXIGRAPH_URL env var.`
I replaced it temporally with "http://localhost:7878" just to test.

Reply:
nah, I forgot to document a step, good spot, I'll add it in 👍